source/features/tag-changelog-link.css

```diff
@@ -0,0 +1,3 @@
+.rgh-tag-changelog-link .release:not(.label-draft) .list-style-none > .d-block:nth-child(3):not(.rgh-changelog-link) {
```

Comment:
What is this targeting? Why 21 exactly? Can we add it via class with JS instead?

Reply:
> What is this targeting?

The third list element in the little sidebar portrayed in the screenshots (so either the "Changelog" dropdown or the "Verified" label). When inserted, the Changelog link will become the third element and the padding will disappear because of `:not(.rgh-changelog-link)`

> Why 21 exactly?

It's the computed height of the link element. Basically a magic number that should be replaced with something better.

> Can we add it via class with JS instead?

Probably, but I thought a pure CSS solution would apply itself faster during page load, and thus alleviate the content jump the most.

Another solution could be to systematically inject an empty list element (or some dummy text) before computing the tags, and then either remove it or replace its content with the link.

Or we could just ignore the content jump (imo it's not that bad) and drop this part altogether.

Reply:
Ohhh I see what you’re doing. You’re avoiding the content jump, so a CSS-only solution is indeed better. However it needs a comment, use margin, and probably a better selector (for example you can target the second element and user margin-bottom instead of using :not)

Reply:
Ohhh I see what you’re doing. You’re avoiding the content jump, so a CSS-only solution is indeed better. However it needs a comment, use margin, and probably a better selector (for example you can target the second element and user margin-bottom instead of using :not)

Reply:
The idea with the `:not()` is to have the selector immediately be disabled when the "Changelog" link is inserted. There's probably a better way to do this, but I haven't found one.
Also I used `padding` because the dropdown already has some margin and we want to add to it, not override it.

Reply:
So it just needs a comment explaining what's going on :D

Reply:
I'd like to get rid of the "magic" `21px`, but I can't find a pure CSS replacement for it.
Any ideas @yakov116?

Reply:
I wont have time until friday to look. But don't count on me finding anything ☕

Reply:
If GitHub had size variables you could use them in a `calc`, but I don't think it does. With JavaScript you could place a whole hidden element while the changelog link loads, but there will be a FOUC.

The hard-coded value is fine as long as you show how you got it. For example include this screenshot
<img width="222" alt="Screen Shot" src="https://user-images.githubusercontent.com/1402241/103394704-bf2d7a80-4aef-11eb-8377-5c59b46cc2b3.png">

Reply:
By the way what you're doing is called "Avoiding layout shift" https://web.dev/cls/